docs/src/manual/models.md

```diff
@@ -62,50 +62,72 @@ functions.
 
 Consult the docstsrings of each attribute for information on what it represents.
 
-## Model attributes
+## ModelLike API
 
-The following attributes are available for models:
+Objects subtyping [`ModelLike`](ref) must implement:
+
+ * [`is_empty`](@ref)
+ * [`empty!`](@ref)
+
+The following attributes are available. Some are required by all
+implementations, and others are optional.
+
+**Required**
+
+ * [`ListOfConstraintAttributesSet`](@ref)
+ * [`ListOfConstraintIndices`](@ref)
+ * [`ListOfConstraintTypesPresent`](@ref)
+ * [`ListOfModelAttributesSet`](@ref)
+ * [`ListOfVariableAttributesSet`](@ref)
+ * [`ListOfVariableIndices`](@ref)
+ * [`NumberOfConstraints`](@ref)
+ * [`NumberOfVariables`](@ref)
+
+**Optional**
 
  * [`Name`](@ref)
  * [`ObjectiveFunction`](@ref)
  * [`ObjectiveFunctionType`](@ref)
  * [`ObjectiveSense`](@ref)
- * [`NumberOfVariables`](@ref)
- * [`ListOfVariableIndices`](@ref)
- * [`ListOfConstraintTypesPresent`](@ref)
- * [`NumberOfConstraints`](@ref)
- * [`ListOfConstraintIndices`](@ref)
- * [`ListOfOptimizerAttributesSet`](@ref)
- * [`ListOfModelAttributesSet`](@ref)
- * [`ListOfVariableAttributesSet`](@ref)
- * [`ListOfConstraintAttributesSet`](@ref)
 
-## Optimizer attributes
-
-The following attributes are available for optimizers:
-
- - [`SolverName`](@ref)
- - [`RawOptimizerAttribute`](@ref)
- - [`Silent`](@ref)
- - [`TimeLimitSec`](@ref)
- - [`NumberOfThreads`](@ref)
- - [`RawSolver`](@ref)
-
-In addition, optimizers should implement the following attributes to provide
-access to solutions:
-
- - [`TerminationStatus`](@ref)
- - [`TerminationStatusCode`](@ref)
- - [`PrimalStatus`](@ref)
- - [`DualStatus`](@ref)
- - [`ResultStatusCode`](@ref)
- - [`RawStatusString`](@ref)
- - [`ResultCount`](@ref)
- - [`ObjectiveValue`](@ref)
- - [`DualObjectiveValue`](@ref)
- - [`ObjectiveBound`](@ref)
- - [`RelativeGap`](@ref)
- - [`SolveTimeSec`](@ref)
- - [`SimplexIterations`](@ref)
- - [`BarrierIterations`](@ref)
- - [`NodeCount`](@ref)
+!!! note
+    To be useful, a [`ModelLike`](@ref) should, in most cases, also implement
```

Comment:
What does "useful" means ? I'd rather say that the above required attributes are the attributes required for a model to be copied from and this below is the required functions to be implemented to be copied to. A read-only model is useful (e.g. it can be created from files  or from matrices), and a write-only as well (it can be stored on files or as matrices for instance).

Reply:
This discussion is difficult because the use-cases are so varied. There is no standard API required by all solvers. I think it's fair to say "in most cases" and leave it at that. We can always revise in future if confusion arises.